test/e2e/framework/job/wait.go

```diff
@@ -196,14 +214,18 @@ func WaitForJobFinish(ctx context.Context, c clientset.Interface, ns, jobName st
 
 // WaitForJobFinishWithTimeout uses c to wait for the Job jobName in namespace ns to finish (either Failed or Complete).
 func WaitForJobFinishWithTimeout(ctx context.Context, c clientset.Interface, ns, jobName string, timeout time.Duration) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, timeout, true, func(ctx context.Context) (bool, error) {
-		curr, err := c.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-
-		return isJobFinished(curr), nil
-	})
+	return framework.Gomega().
+		Eventually(ctx, framework.RetryNotFound(framework.GetObject(c.BatchV1().Jobs(ns).Get, jobName, metav1.GetOptions{}))).
```

Comment:
I would use `HandleRetry` here instead of `RetryNotFound` and just retry transient errors. 

IMO the test writer expects the job to be there (otherwise she wouldn't for it to finish). If it's not, there's clearly an error that shouldn't be retried.

Reply:
When I implement it, I have the same question. But when I look at other code in this file, I don't know the reason why it will try to get the job when it is not found, so I follow the same pattern.

I think we can remove the `framework.RetryNotFound` from https://github.com/carlory/kubernetes/blob/master/test/e2e/framework/job/wait.go#L259.